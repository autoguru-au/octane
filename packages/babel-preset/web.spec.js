process.env.NODE_ENV = 'production';

const { transformAsync } = require('@babel/core');

const fixture = `
const myVar = new Map([]);
class TestClass {
	static staticProp = 'static prop';
	prop = 'prop';
	
	get getterProp() {
		return 'getter prop';
	}
}

const objectA = {a: 'a', b: 'b'};

const d = 'd';
const objectB = {...objectA, c: 'c', [d]: 'd'};
`;

describe('web preset', function() {
	it('should match the snapshot', async () => {
		const { code } = await transformAsync(fixture, {
			babelrc: false,
			presets: [[require('./web'), { debug: false }]],
		});

		return expect(code).toMatchSnapshot();
	});

	it('should allow for corejs override', async () => {
		const { code } = await transformAsync(fixture, {
			babelrc: false,
			presets: [[require('./web'), { debug: false, corejs: 3 }]],
		});

		return expect(code).toMatchSnapshot();
	});
});
