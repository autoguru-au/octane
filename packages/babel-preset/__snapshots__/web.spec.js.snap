// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`web preset should allow for corejs override 1`] = `
"\\"use strict\\";

import _Object$assign from \\"@babel/runtime-corejs3/core-js-stable/object/assign\\";
import _createClass from \\"@babel/runtime-corejs3/helpers/esm/createClass\\";
import _Map from \\"@babel/runtime-corejs3/core-js-stable/map\\";

var _Object$assign2;

var myVar = new _Map([]);

var TestClass =
/*#__PURE__*/
function () {
  function TestClass() {
    this.prop = 'prop';
  }

  _createClass(TestClass, [{
    key: \\"getterProp\\",
    get: function get() {
      return 'getter prop';
    }
  }]);

  return TestClass;
}();

TestClass.staticProp = 'static prop';
var objectA = {
  a: 'a',
  b: 'b'
};
var d = 'd';

var objectB = _Object$assign({}, objectA, (_Object$assign2 = {
  c: 'c'
}, _Object$assign2[d] = 'd', _Object$assign2));"
`;

exports[`web preset should match the snapshot 1`] = `
"\\"use strict\\";

import _Object$assign from \\"@babel/runtime-corejs2/core-js/object/assign\\";
import _createClass from \\"@babel/runtime-corejs2/helpers/esm/createClass\\";
import _Map from \\"@babel/runtime-corejs2/core-js/map\\";

var _Object$assign2;

var myVar = new _Map([]);

var TestClass =
/*#__PURE__*/
function () {
  function TestClass() {
    this.prop = 'prop';
  }

  _createClass(TestClass, [{
    key: \\"getterProp\\",
    get: function get() {
      return 'getter prop';
    }
  }]);

  return TestClass;
}();

TestClass.staticProp = 'static prop';
var objectA = {
  a: 'a',
  b: 'b'
};
var d = 'd';

var objectB = _Object$assign({}, objectA, (_Object$assign2 = {
  c: 'c'
}, _Object$assign2[d] = 'd', _Object$assign2));"
`;
